keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -175,7 +172,7 @@ class WordPieceTokenizer(tokenizer.Tokenizer):
 
     def __init__(
         self,
-        vocabulary: Union[Iterable[str], str] = None,
+        vocabulary=None,
```

Comment:
I think that's ok! Type annotations can be partial by design.

I'm fine either way here. From discussion on chat, my understanding was we would like to avoid stuff like this (union types). [Here's](https://keras.io/api/keras_nlp/tokenizers/word_piece_tokenizer/) how they look like in docs. In some ways I don't mind. It is nice to be able to tell at a glance that the vocabulary can be a string or list of strings.

But I also think only doing type annotations for basic types is a reasonable approach, and what I though we were heading towards right now.

@fchollet thoughts?